wallet/loader.go

```diff
@@ -108,8 +108,37 @@ func (l *Loader) RunAfterLoad(fn func(*Wallet)) {
 func (l *Loader) CreateNewWallet(pubPassphrase, privPassphrase, seed []byte,
 	bday time.Time) (*Wallet, error) {
 
+	defer l.mu.Unlock()
+	l.mu.Lock()
+
+	if l.wallet != nil {
+		return nil, ErrLoaded
+	}
+
+	db, err := l.createLocalDB()
```

Comment:
Alternatively, what if we jsut added the database as new argument to `NewLoader` where we take all the other non-wallet specific data? This would reduce the size of the diff, as we'd just need to check if that field is nil or not. If we don't want to explicitly break the API here, than we can use a functional argument, which also opens up the way for us extending the methods further in the future.